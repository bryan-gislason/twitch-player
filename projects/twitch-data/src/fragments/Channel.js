import gql from "graphql-tag";
import BroadcasterFragment from "./Broadcaster";
import CategoryFragment from "./Category";
import TagFragment from "./Tag";

export default gql`
  fragment Channel on Stream {
    id
    key: id
    previewImageURL
    viewersCount
    broadcaster {
      ...Broadcaster
    }
    category: game {
      ...Category
    }
    tags {
      ...Tag
    }
  }
  ${BroadcasterFragment}
  ${CategoryFragment}
  ${TagFragment}
`;
