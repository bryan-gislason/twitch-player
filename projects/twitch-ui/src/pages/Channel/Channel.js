import PropTypes from "prop-types";
import React, { memo } from "react";
import { StyleSheet, View } from "react-native";
import ChannelPlayerFrame from "../../components/ChannelPlayerFrame";

const Channel = ({ match, testID = "channel-page" }) => {
  const { name } = match.params;

  return (
    <View style={styles.root} testID={testID}>
      <ChannelPlayerFrame name={name} />
    </View>
  );
};

Channel.propTypes = {
  match: PropTypes.object.isRequired,
  testID: PropTypes.string,
};

const styles = StyleSheet.create({
  root: {
    flex: 1,
  },
});

export default memo(Channel);
