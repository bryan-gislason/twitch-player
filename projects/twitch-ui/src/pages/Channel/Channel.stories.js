import { storiesOf } from "@storybook/react";
import React from "react";
import Main from "../../components/Main";
import Channel from ".";

storiesOf("Pages|Channel", module).add("default", () => (
  <Main>
    <Channel match={{ params: { name: "twitchpresents" } }} />
  </Main>
));
