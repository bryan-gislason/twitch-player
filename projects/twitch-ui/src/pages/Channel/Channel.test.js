import React from "react";
import { render } from "../../testing";
import Channel from "./Channel";

jest.mock("../../components/ChannelPlayerFrame", () => "mock-channel-player");

describe("Channel", () => {
  let instance;

  beforeEach(() => {
    instance = render(
      <Channel match={{ params: { name: "twitchpresents" } }} />
    );
  });

  it("renders the page", () => {
    expect(instance.getByTestId("channel-page")).toMatchSnapshot();
  });
});
