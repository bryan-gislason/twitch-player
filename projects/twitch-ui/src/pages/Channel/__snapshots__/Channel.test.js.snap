// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Channel renders the page 1`] = `
<div
  class="rn-alignItems-1oszu61 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexGrow-16y2uox rn-flexShrink-1wbh5a2 rn-flexBasis-1ro0kt6 rn-flexDirection-eqz5dr rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-zIndex-1lgpqti"
  data-testid="channel-page"
>
  <iframe
    allowfullscreen=""
    data-testid="channel-player"
    src="https://player.twitch.tv/?channel=twitchpresents"
    style="border: 0px; flex: 1;"
  />
</div>
`;
